vendor/github.com/prometheus/prometheus/pkg/labels/matcher.go

```diff
@@ -46,8 +46,7 @@ func NewMatcher(t MatchType, n, v string) (*Matcher, error) {
 		Value: v,
 	}
 	if t == MatchRegexp || t == MatchNotRegexp {
-		m.Value = "^(?:" + v + ")$"
-		re, err := regexp.Compile(m.Value)
+		re, err := regexp.Compile("^(?:" + m.Value + ")$")
```

Comment:
I would prefer not to diverge from vendored packages.

I went through the code and made a couple changes, but I'll leave it to you to apply them to your branch.

First, reverting the change to the vendored package:
```
git revert 5d0bba4b3b3e682b4eae10381ce96660d5c3979a
```

Then here's a patch file that you can apply. Read through it to see what it does, but basically it separates out the actual parsing code in `pkg/parse` so that there's an exported function for parsing the raw input string, which the CLI then uses to send to the API.

[parse_regex.txt](https://github.com/prometheus/alertmanager/files/916562/parse_regex.txt)

Note: Apparently github won't let me upload a `.patch` file, so it says text, but you should be able to apply it all the same.